calypso/csadmin/clicontracts/write.go

```diff
@@ -0,0 +1,140 @@
+package clicontracts
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"go.dedis.ch/cothority/v3"
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/cothority/v3/byzcoin/bcadmin/lib"
+	"go.dedis.ch/cothority/v3/calypso"
+	"go.dedis.ch/cothority/v3/darc"
+	"go.dedis.ch/protobuf"
+	"gopkg.in/urfave/cli.v1"
+)
+
+// WriteSpawn creates a new instance of a write contract. It expects a public
+// key point in hex string format, which is the collective public key generated
+// by the DKG. The data that will be encrypted under the collective public key
+// is provided as a string and then converted as a slice of bytes. If everything
+// goes well, it prints the instance id of the newly spawned Write instance.
+// With the --export option, the instance id is sent to STDOUT.
+func WriteSpawn(c *cli.Context) error {
+	bcArg := c.String("bc")
+	if bcArg == "" {
+		return errors.New("--bc flag is required")
+	}
+
+	cfg, cl, err := lib.LoadConfig(bcArg)
+	if err != nil {
+		return err
+	}
+
+	dstr := c.String("darc")
+	if dstr == "" {
+		dstr = cfg.AdminDarc.GetIdentityString()
+	}
+	d, err := lib.GetDarcByString(cl, dstr)
+	if err != nil {
+		return err
+	}
+
+	data := c.String("data")
+	if data == "" {
+		return errors.New("please provide data with --data")
```

Comment:
i think it'll be quite convenient to allow reading data from stdin if the `--data` flag is missing, e.g., when you want to put a file on calypso

Reply:
Actually I'm realising this should not be called "data" and it may confuse the reader. The "data" I mention here is actually the symmetric key that is embedded into a `kyber.Point`. This has a limited space of [29 bit](https://github.com/dedis/kyber/blob/master/group/edwards25519/point.go#L107) and the user should not use it to store a file.

Instead, there is actually a `data` field that is used to stored the encrypted data. So I should rename "--data" to "--key" and add a "--data" that can be read from STDIN or given in argument and that is stored on the actual `data` field of the Writer contract. But this content has to be encrypted by the user.